src/ParameterDistributions.jl

```diff
@@ -69,19 +72,23 @@ end
 
 # For the transforms
 abstract type ConstraintType end
+abstract type NoConstraint <: ConstraintType end
+abstract type BoundedBelow <: ConstraintType end
+abstract type BoundedAbove <: ConstraintType end
+abstract type Bounded <: ConstraintType end
 
 """
     Constraint <: ConstraintType
 
 Class describing a 1D bijection between constrained and unconstrained spaces.
 """
-struct Constraint <: ConstraintType
+struct Constraint{T} <: ConstraintType
     constrained_to_unconstrained::Function
     c_to_u_jacobian::Function
     unconstrained_to_constrained::Function
+    parameters::Union{Dict, Nothing}
```

Comment:
Can you add docstrings on top of each element of this struct for the API docs?

Also, I think we might be overusing the word `parameters` here, since these are the `parameters defining a parameter transformation`. Could we use another term, perhaps `bounds`?

Reply:
It is any parameter that could be used to build a constraint - currently they are bounds but it could also be e.g. a scaling parameter in the sigmoid. So may not only be bounds. I can definitely rename though

Reply:
I think I'm happy to use `bounds` and change the term if we find a good use-case for when the parameter value is not a bound